src/dune_rules/foreign.ml

```diff
@@ -92,6 +92,7 @@ module Stubs = struct
     type t =
       | Dir of String_with_vars.t
       | Lib of Loc.t * Lib_name.t
+      | Include of Loc.t * Ordered_set_lang.Unexpanded.t
```

Comment:
From what I understand, `(include ..)` is allowed to expanded into OSL, right? That seems counterintuitive to me because OSL isn't accepted in the field when it is statically interpreted. Furthermore, the following isn't allowed either:

```
$ cat foo
  (baz (lib foo))
$ cat >dune << EOF
> ..
> (include_dirs (include foo))
> EOF
```
Which I would expect to be allowed.

Reply:
Good point. I think it was a mistake to use OSL for this feature. I'll change so that the file specified to `(include ...)` can contain a sexp in the same notation as the `(include_dirs ...)` field.

Reply:
Yes indeed. It was a nice idea, but it doesn't seem to work out in practice.

Reply:
I've updated this PR such that now the contents of the included file is parsed using the `(include_dir ...)` parser